src/RRule.php

```diff
@@ -2324,4 +2339,27 @@ public function humanReadable(array $opt = array())
 		return $str;
 	}
 
+	/**
+	 * Attempt to fetch a locale either from Intl or setlocale()
+	 *
+	 * @return null|string
+	 */
+	private function getSystemLocale()
+	{
+		$locale = null;
+
+		if ( self::$intl_loaded ) {
+			$locale = \Locale::getDefault();
+		} else {
+			$locales = setlocale(LC_ALL, 0);
```

Comment:
The problem with `setlocale()` is that it returns `en_US.utf-8` as a locale, so I still need to grep out the `en_US` part. It's possible to make `i18nLoad()`'s preg_match a little bit looser, so that it allows for such locales, since it still extracts what it needs from the locale string.